src/taco-utils/telemetryHelper.ts

```diff
@@ -246,6 +246,38 @@ module TacoUtility {
                 () => generator.send()); // After
         }
 
+        /**
+         * Creates an object that can be passed to external modules to report telemetry events.
+         * @param componentName - a name to prefix on the telemetry event names received from the external module.
+         * @param baseProps - a collection of properties to include with all events.
+         * @param errorHandler - an optional function to wrap or modify any errors (wrap in a TacoError, for example).
+         * @returns {{sendTelemetry: (function(string, ITelemetryProperties, Error=): void)}}
+         */
+        public static getExternalTelemetryObject(componentName: string, baseProps: Telemetry.ITelemetryProperties, errorHandler?: (error: Error) => Error) : Object {
```

Comment:
The reason I didn't is that this is just being passed to some external (non-TACO) module, so there didn't seem to be any benefit. I actually originally had a comment to that effect, but must have deleted it when moving code around. Can you think of a scenario where having an interface defined would be useful? I guess if that external module was written in TypeScript it would be useful (currently it is only being used by JavaScript code).